akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -666,24 +666,34 @@ private[akka] class ActorSystemImpl(
 
             if (settings.JvmExitOnFatalError) {
               try {
-                markerLogging.error(LogMarker.Security, cause, "Uncaught error from thread [{}] shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled", thread.getName)
-                import System.err
-                err.print("Uncaught error from thread [")
-                err.print(thread.getName)
-                err.print("] shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[")
-                err.print(name)
-                err.println("]")
-                cause.printStackTrace(System.err)
-                System.err.flush()
+                logFatalError("shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for", cause, thread)
               } finally {
                 System.exit(-1)
               }
             } else {
-              markerLogging.error(LogMarker.Security, cause, "Uncaught fatal error from thread [{}] shutting down ActorSystem [{}]", thread.getName, name)
+              logFatalError("shutting down", cause, thread)
               terminate()
```

Comment:
`logFatalError` is pretty simple, if that crashes I'm not sure there's much of a chance `terminate()` will do anything useful... might make the code a bit more symmetrical though :)